src/app/pages/blog/article-summary/article-summary.jsx

```diff
@@ -0,0 +1,73 @@
+import React, {useState, useRef, useEffect} from 'react';
+import {RawHTML} from '~/components/jsx-helpers/jsx-helpers.jsx';
+import {BylineJsx} from '~/components/byline/byline';
+import $ from '~/helpers/$';
+import debounce from 'lodash/debounce';
+
+export function blurbModel(articleSlug, data) {
+    if (!data) {
+        return {};
+    }
+    return {
+        headline: data.heading,
+        subheading: data.subheading,
+        image: data.article_image,
+        altText: data.article_image_alt,
+        body: data.body_blurb,
+        author: data.author,
+        date: data.date,
+        articleSlug
+    };
+}
+
+export function ArticleSummary({
+    articleSlug, altText, image, headline, subheading, body, date, author,
+    forwardRef={}, setPath
+}) {
+    function onClick(event) {
+        event.preventDefault();
+        setPath(event.target.href);
+    }
+
+    return (
+        <React.Fragment>
+            <a className="link-image" href={`/blog/${articleSlug}`}
+                ref={forwardRef}
+                aria-label={altText}
+                style={`background-image: url("${image}")`}></a>
+            <div className="text-block">
+                <h2 className="article-headline">
+                    <a href={`/blog/${articleSlug}`} onClick={onClick}>{headline}</a>
+                </h2>
+                {
+                    subheading &&
+                        <h3>{subheading}</h3>
+                }
+                <RawHTML className="article-blurb" html={body} />
+                <a className="go-to" href={`/blog/${articleSlug}`} onClick={onClick}>read more</a>
+                <BylineJsx date={date} author={author} />
+            </div>
+        </React.Fragment>
+    );
+}
+
+export default function DelayedImagesSummary({image, ...otherProps}) {
+    const [delayedImage, setDelayedImage] = useState('');
+    const elRef = useRef();
+
+    useEffect(() => {
+        const handleScroll = debounce(() => {
+            if (elRef.current && $.overlapsViewport(elRef.current)) {
+                window.removeEventListener('scroll', handleScroll);
+                setDelayedImage(image);
+            }
+        }, 200);
+
+        window.addEventListener('scroll', handleScroll);
+        handleScroll();
+    }, [image]);
```

Comment:
You'll need to return a function that will remove the event listener on unmount

this is also perfect candidate to make a custom `usesScrollPosition` hook.   https://dev.to/n8tb1t/tracking-scroll-position-with-react-hooks-3bbj is a decent walkthrough on that.  

Then for a more advanced one you could make the hook only add a single listener that would notify all the images when it's scrolled.  It might not be worth it to do that now though, but would be a good optimization

Reply:
Since I'm looking at the scroll position of the window object, I implemented a Context for window information (currently only scrollY, but in theory more things could be put in it). That gives me one scroll listener that notifies everybody. I'd appreciate your doing a sanity check on it.